sycl/include/CL/sycl/kernel.hpp

```diff
@@ -202,11 +202,8 @@ class __SYCL_EXPORT kernel {
 
   template <backend Backend>
   __SYCL_DEPRECATED("Use SYCL 2020 sycl::get_native free function")
-  typename backend_traits<Backend>::template return_type<kernel> get_native()
-      const {
-    return detail::pi::cast<
-        typename backend_traits<Backend>::template return_type<kernel>>(
-        getNativeImpl());
+  auto get_native() const -> backend_return_t<Backend, kernel> {
+    return detail::pi::cast<backend_return_t<Backend, kernel>>(getNativeImpl());
```

Comment:
Just out of curiosity: why kernel uses `getNativeImpl`, when all other classes use `getNative`?

Reply:
The `getNativeImpl` method is renamed into the `getNative` as the method is named in all other classes.